### PR TITLE
[FE/BE] Update redirect routing

### PR DESCRIPTION
**Pull request by stonefruit**

### CODE REVIEW

**Comment by stonefruit** [comment]
Prettier done on first commit, view second commit for changes.

File: frontend/api/createAxiosInstance.ts [hunk 1]
```diff
@@ -1,13 +1,30 @@
...
+		(error) => {
+			const STATUS_UNAUTHORIZED = 401;
+			if (error?.response?.status === STATUS_UNAUTHORIZED) {
+				// Server's 302 redirection only works on documents, and not xhr calls.
+				const path =
+					error?.response?.data?.redirectPath ?? "/shelter/login";
+				console.log(`Unauthenticated API call. Redirecting to ${path}`);
+				window.location.assign(path);
+			}
+			return error;
```
Comment (qroll) [inline frontend/api/createAxiosInstance.ts:23]: returning here seems to swallow errors; the promise resolves with the error and the caller will continue as though the request was successful

File: frontend/pages/shelter/login.tsx [hunk 2]
```diff
@@ -15,8 +16,10 @@ const ShelterLogin = () => {
 			const axios = createAxiosInstance();
 			const {
 				data: { payload },
-			} = await axios.post("/api/login", values);
-			window.location.assign("/shelter/home");
+			} = await axios.post<LoginResponse>("/api/login", values);
+			if (payload.shelterId) {
+				window.location.assign("/shelter/home");
+			}
```
Comment (qroll) [inline frontend/pages/shelter/login.tsx:22]: might not be in the scope of your PR, but what's supposed to happen when there's no shelter id?

**Review by qroll — CHANGES_REQUESTED** [review]

File: frontend/api/createAxiosInstance.ts (hunk 1 above)
Comment (stonefruit) [reply frontend/api/createAxiosInstance.ts:23]: I see. no wonder the axios example used Promise.reject. changing it.

File: frontend/pages/shelter/login.tsx (hunk 2 above)
Comment (stonefruit) [reply frontend/pages/shelter/login.tsx:22]: not sure yet. we need to decide on the home page for adopters.

**Review by qroll — APPROVED** [review]

**Pull request merged**